app/config/services/system.xml

```diff
@@ -65,6 +65,13 @@
             <tag name="kernel.event_listener" event="security.interactive_login" method="onSecurityInteractiveLogin" />
         </service>
 
+        <service id="AppBundle\Security\InactiveAdminListener">
+            <argument type="service" id="service_container"/>
```

Comment:
Does it work with `container` only instead of `service_container`? I think you may inject the whole container whereas we need a service locator (a subset of it).